Archive/Imo/README.md

```diff
@@ -0,0 +1,48 @@
+# IMO problems
+
+We have a collection of solutions to IMO problems in mathlib, stored under `/Archive/Imo/`.
+
+These are part of mathlib for three purposes:
+* The [IMO Grand Challenge](https://imo-grand-challenge.github.io/) will need training data, and exemplars,
+  and this is a reasonable place to collect Lean samples of IMO problems.
+* As with the rest of `/Archive/`, we want to have high quality examples
```

Comment:
```suggestion
* As with the rest of `Archive/`, we want to have high quality examples
```
seems more correct to me (like `Mathlib/` in another file), because you'd `cd Archive/` rather than `cd /Archive/` when you're in mathlib's root directory. (4 occurrences)